pkg/util/modules/modules.go

```diff
@@ -0,0 +1,157 @@
+package modules
+
+import (
+	"fmt"
+
+	"github.com/cortexproject/cortex/pkg/util/services"
+	"github.com/pkg/errors"
+)
+
+type service func() (services.Service, error)
```

Comment:
`service` is not a good name. I'm not sure extra type is necessary, it's used at exactly two places.